### PR TITLE
[MSGINA] Make DLG_DATA object-oriented

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

JIRA issue: [CORE-16697](https://jira.reactos.org/browse/CORE-16697)

- Rename `hBitmap` as `hLogoBitmap` of `DLG_DATA` structure.
- Make `DLG_DATA` object-oriented.
- Add `DlgData_Create`. `DlgData_LoadBitmaps`, and `DlgData_Destroy` functions.

### CODE REVIEW

**Comment by Extravert-ir** [comment]
Can you please add more description in commit message?
And add jira ticket ID there

File: dll/win32/msgina/gui.c [hunk 1]
```diff
@@ -30,14 +31,62 @@ typedef struct _LEGALNOTICEDATA
...
     DWORD BarWidth;
     DWORD BarHeight;
 } DLG_DATA, *PDLG_DATA;
 
+static PDLG_DATA
+DlgData_Create(HWND hwndDlg)
+{
+    PDLG_DATA pDlgData;
+    pDlgData = HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY, sizeof(DLG_DATA));
+    if (pDlgData)
```
Comment (SergeGautherie) [inline dll/win32/msgina/gui.c]: Merge these 2 lines?

File: dll/win32/msgina/gui.c [hunk 2]
```diff
@@ -30,14 +31,62 @@ typedef struct _LEGALNOTICEDATA
...
+    {
+        SetWindowLongPtrW(hwndDlg, GWLP_USERDATA, (LONG_PTR)pDlgData);
+    }
+    return pDlgData;
+}
+
+static BOOL
+DlgData_LoadBitmaps(PDLG_DATA pDlgData)
+{
+    BITMAP bm;
```
Comment (SergeGautherie) [inline dll/win32/msgina/gui.c]: Add a blank line?

File: dll/win32/msgina/gui.c [hunk 3]
```diff
@@ -30,14 +31,62 @@ typedef struct _LEGALNOTICEDATA
...
+    DWORD LogoHeight;
     DWORD BarWidth;
     DWORD BarHeight;
 } DLG_DATA, *PDLG_DATA;
 
+static PDLG_DATA
+DlgData_Create(HWND hwndDlg)
+{
+    PDLG_DATA pDlgData;
+    pDlgData = HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY, sizeof(DLG_DATA));
```
Comment (SergeGautherie) [inline dll/win32/msgina/gui.c]: Merge these 2 lines?

File: dll/win32/msgina/gui.c [hunk 4]
```diff
@@ -30,14 +31,62 @@ typedef struct _LEGALNOTICEDATA
...
+    pDlgData->LogoWidth = bm.bmWidth;
+    pDlgData->LogoHeight = bm.bmHeight;
+
+    pDlgData->hBarBitmap = LoadImageW(hDllInstance, MAKEINTRESOURCEW(IDI_BAR),
+                                      IMAGE_BITMAP, 0, 0, LR_DEFAULTCOLOR);
+    GetObject(pDlgData->hBarBitmap, sizeof(bm), &bm);
+    pDlgData->BarWidth = bm.bmWidth;
+    pDlgData->BarHeight = bm.bmHeight;
+
+    return pDlgData->hLogoBitmap && pDlgData->hBarBitmap;
```
Comment (Fraizeraust) [inline dll/win32/msgina/gui.c]: There is some confusion here. Your `DlgData_LoadBitmaps()` is defined as a **BOOL** yet you return two handles of logo and bar image bitmaps. Did I miss something (is it correct in C)?

Comment (HBelusca) [reply dll/win32/msgina/gui.c]: The result will be TRUE if both hLogoBitmap and hBarBitmap are non-NULL. However if any of these is NULL then the returned value will be FALSE.

File: dll/win32/msgina/gui.c [hunk 5]
```diff
@@ -30,14 +31,62 @@ typedef struct _LEGALNOTICEDATA
...
+    DWORD LogoWidth;
+    DWORD LogoHeight;
     DWORD BarWidth;
     DWORD BarHeight;
 } DLG_DATA, *PDLG_DATA;
 
+static PDLG_DATA
+DlgData_Create(HWND hwndDlg)
+{
+    PDLG_DATA pDlgData = HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY, sizeof(DLG_DATA));
```
Comment (SergeGautherie) [inline dll/win32/msgina/gui.c]: `sizeof(*pDlgData)`?

**Comment by Fraizeraust** [comment]
Your newly pushed commit brought a regression -- the bitmap bar is no longer loaded. I'll provide a patch ASAP.

**Pull request merged**